src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -117,14 +160,21 @@ public boolean equals(Object other) {
         }
 
         EditCommand otherEditCommand = (EditCommand) other;
-        return index.equals(otherEditCommand.index)
-                && editPersonDescriptor.equals(otherEditCommand.editPersonDescriptor);
-    }
 
+        // Check if both the name and nric are equal for equality check
+        boolean isNameEqual = (name == null && otherEditCommand.name == null)
+                || (name != null && name.equals(otherEditCommand.name));
+
+        boolean isNricEqual = (nric == null && otherEditCommand.nric == null)
+                || (nric != null && nric.equals(otherEditCommand.nric));
+
```

Comment:
I made a similar mistake here, this can be replaced with the `Objects.equals` method that Java provides you.